lib/tests/anyOf-matches-json-schema.test.ts

```diff
@@ -0,0 +1,78 @@
+import { getZodSchema } from "../src/openApiToZod";
+import { test, expect } from "vitest";
+
+test("anyOf-matches-json-schema", () => {
+    expect(
+        getZodSchema({
+            // @see: https://swagger.io/docs/specification/data-models/oneof-anyof-allof-not/#anyof
+            schema: {
+                anyOf: [
+                    {
+                        type: "object",
+                        properties: {
+                            age: {
+                                type: "integer",
+                            },
+                            nickname: {
+                                type: "string",
+                            },
+                        },
+                        required: ["age"],
+                    },
+                    {
+                        type: "object",
+                        properties: {
+                            pet_type: {
+                                type: "string",
+                                enum: ["Cat", "Dog"],
+                            },
+                            hunts: {
+                                type: "boolean",
+                            },
+                        },
+                        required: ["pet_type"],
+                    },
+                ],
+            },
+        })
+    ).toMatchInlineSnapshot(
+        '"z.union([z.object({ age: z.number().int(), nickname: z.string().optional() }).passthrough(), z.object({ pet_type: z.enum(["Cat", "Dog"]), hunts: z.boolean().optional() }).passthrough()])"'
+    );
+
+    expect(
+        getZodSchema({
+            schema: {
+                anyOf: [
+                    {
+                        type: "object",
+                        properties: {
+                            age: {
+                                type: "integer",
+                            },
+                            nickname: {
+                                type: "string",
+                            },
+                        },
+                        required: ["age"],
+                    },
+                    {
+                        type: "object",
+                        properties: {
+                            pet_type: {
+                                type: "string",
+                                enum: ["Cat", "Dog"],
+                            },
+                            hunts: {
+                                type: "boolean",
+                            },
+                        },
+                        required: ["pet_type"],
+                    },
+                    { type: "number" },
+                ],
+            },
+        })
+    ).toMatchInlineSnapshot(
+        '"z.union([z.object({ age: z.number().int(), nickname: z.string().optional() }).passthrough(), z.object({ pet_type: z.enum(["Cat", "Dog"]), hunts: z.boolean().optional() }).passthrough(), z.number()])"'
+    );
+});
```

Comment:
there's no way to do that atm, although it would be very useful, but I guess for now you could just copy/paste the result and re-use that in another test with a manual call to the zod parse fn

Reply:
that works! i'll do it tomorrow, just finishing up for the day :)

Reply:
not sure if this is too cursed to commit?

```ts
function createValidator(zodSchema: CodeMeta) {
    return new Function(
        "input",
        `
        with(this) {
            return ${zodSchema}.parse(input)
        }
    `
    ).bind({ z });
}

const zodSchema = getZodSchema({ schema: { type: "number" } })
const validator = createValidator(zodSchema)

expect(() => validator("string")).toThrow()
```

Reply:
it's easier to beg for forgiveness than ask permission, so i just pushed this haha. it's a quick (and definitely dirty!) way to validate the behavior of schemas produced, but i think it's valuable here. simple to back out if you'd rather not. let me know!

Reply:
I'm not sure what the `with` keyword does tbh, it's been a while since I saw this 😅  but yeah overall this seems fine, it does the job ! keep it

Reply:
it looks great !